### PR TITLE
enable fp8 cast for inductor CPU

### PR DESCRIPTION
**Pull request by zhuhaozhe**
Enable FP8 cast for this issue https://github.com/pytorch/pytorch/issues/117119.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #117737

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

File: test/inductor/test_cpu_repro.py [hunk 1]
```diff
@@ -2765,6 +2771,23 @@ def forward(self, x):
...
+            return y0, y1
+
+        compiled_fp8_cast = torch.compile(fp8_cast, backend="inductor", dynamic=True)
+
+        shape = [int(dim) for dim in shape.split(",")]
+        x = torch.rand(*shape, device="cpu", dtype=dtype)
+        y0_fp8, y1_fp8 = compiled_fp8_cast(x)
+
+        torch.testing.assert_close(y0_fp8, x, rtol=5e-1, atol=5e-1)
+        torch.testing.assert_close(y1_fp8, x, rtol=5e-1, atol=5e-1)
```
Comment (jgong5) [inline test/inductor/test_cpu_repro.py:2789]: why not just comparing with eager like other tests do?

**Review by jgong5 — APPROVED** [review]

File: test/inductor/test_cpu_repro.py (hunk 1 above)
Comment (zhuhaozhe) [reply test/inductor/test_cpu_repro.py]: I am copying it from the `CUDA` test before. Now I aligned it with other tests now (using `self.common`).

**Review by jgong5 — CHANGES_REQUESTED** [review]

**Review by jgong5 — APPROVED** [review]

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**